packages/server-wallet/src/wallet/index.ts

```diff
@@ -210,47 +224,54 @@ const takeActions = async (channels: Bytes32[]): Promise<ExecutionResult> => {
   const channelResults: ClientChannelResult[] = [];
   let error: Error | undefined = undefined;
   while (channels.length && !error) {
-    const tx = await knex.transaction();
-
-    const setError = async (e: Error): Promise<void> => {
-      error = e;
-      await tx.rollback();
-    };
-    const markChannelAsDone = async (): Promise<any> => {
-      channels.shift() as string;
-    };
-
-    const handleAction = async (action: ProtocolAction): Promise<any> => {
-      switch (action.type) {
-        case 'SignState': {
-          const {outgoing, channelResult} = await Store.signState(action.channelId, action, tx);
-          outgoing.map(n => outbox.push(n.notice));
-          channelResults.push(channelResult);
-          return;
+    await knex.transaction(async tx => {
+      const setError = async (e: Error): Promise<void> => {
+        error = e;
+        await tx.rollback(error);
+      };
+      const markChannelAsDone = async (): Promise<any> => {
+        channels.shift() as string;
+      };
```

Comment:
This is neither async and returns `void`, maybe we can update the signature?

Reply:
It has the specific signature for use in `Option.fold`. We should simply not use `Option.fold`, which would correct this.

Can be done with https://github.com/statechannels/statechannels/issues/2379.